components/script/layout_dom/node.rs

```diff
@@ -76,6 +76,11 @@ impl<'dom> ServoLayoutNode<'dom> {
         ServoLayoutNode { node: n }
     }
 
+    /// Create a new [`ServoLayoutNode`] for this given [`TrustedNodeAddress`].
+    ///
+    /// # Safety
+    ///
+    /// The address pointed to by `address` should point to a valid node in memory.
```

Comment:
This seems redundant given that the documenation for `TrustedNodeAddress` [claims that it represents a valid Node address](https://github.com/servo/servo/blob/c430645606396a300b57a4d9c909db959486ca4c/components/shared/script_layout/lib.rs#L141). Perhaps this is still needed since the field in TrustedNodeAddress is public, but I wonder if that should be made private?

Also, are there no other conditions that need to be highlighted in this safety section? As an example (I'm not really familiar with this code), is there anything to be said about whether or not the DOM node is already rooted?

Reply:
Unfortunately, I don't think we can make those kind of guarantees as `TrustedNodeAddress` is sent over IPC channels. This is all very bad, of course, and we should remove `TrustedNodeAddress` entirely, but that's a much larger task.